### PR TITLE
Send meetingStatus to ingestion service

### PR DESCRIPTION
**Pull request by dylonChime**
## ℹ️ Description
Route meetingStatus to ingestion service for meeting start and failed events.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [ ] README update
    - [ ] CHANGELOG update
    - [ ] guides update
- [ ] This change requires a dependency update
    - [ ] Amazon Chime SDK Media
    - [ ] Other (update corresponding legal documents)

## 🧪 How Has This Been Tested?
Tested the demo and checked ingestion logs

### Additional Manual Test
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Rotate screen back and forth

## 📱 Screenshots, if available
*provide screenshots/video record if there's a UI change in demo app*

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dingyishen-amazon — APPROVED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift [hunk 1]
```diff
@@ -107,7 +107,7 @@ extension DefaultAudioClientController: AudioClientController {
         audioClientObserver.notifyAudioClientObserver { (observer: AudioVideoObserver) in
             observer.audioSessionDidStartConnecting(reconnecting: false)
         }
-        eventAnalyticsController.publishEvent(name: .meetingStartRequested)
+        eventAnalyticsController.publishEvent(name: .meetingStartRequested, attributes: [EventAttributeName.meetingStatus: audioClientObserver.audioStatus])
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:110]: Where is `audioStatus` from? I don't see it in AudioClientObserver.

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift [hunk 2]
```diff
@@ -141,7 +141,7 @@ extension DefaultAudioClientController: AudioClientController {
             Self.state = .started
         } else {
             logger.error(msg: "Cannot start audio client with status = \(status.rawValue)")
-            eventAnalyticsController.publishEvent(name: .meetingStartFailed)
+            eventAnalyticsController.publishEvent(name: .meetingStartFailed, attributes: [EventAttributeName.meetingStatus: audioClientObserver.audioStatus])
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:144]: ditto

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift (hunk 1 above)
Comment (georgezy-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:110]: Nvm, you just added it.

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientObserver.swift [hunk 3]
```diff
@@ -348,9 +348,9 @@ class DefaultAudioClientObserver: NSObject, AudioClientDelegate {
         }
     }
 
-    private func notifyStartSucceeded() {
+    private func notifyStartSucceeded(newAudioStatus: MeetingSessionStatusCode) {
         meetingStatsCollector.updateMeetingStartTimeMs()
-        eventAnalyticsController.publishEvent(name: .meetingStartSucceeded)
+        eventAnalyticsController.publishEvent(name: .meetingStartSucceeded, attributes: [EventAttributeName.meetingStatus: newAudioStatus])
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientObserver.swift:353]: Won't the status be always `OK` for success?

Comment (dylonChime) [reply AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientObserver.swift:353]: Yes, but it makes sense to provide it. `meetingStatus` is independent of the operation so it makes searching ingestion logs more intuitive.

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift (hunk 2 above)
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:144]: Let's also update unit tests.

**Review by georgezy-amzn — DISMISSED** [review]

**Review by georgezy-amzn — DISMISSED** [review]

**Review by georgezy-amzn — APPROVED** [review]

**Review by ziyiz-amzn — DISMISSED** [review]

**Review by georgezy-amzn — APPROVED** [review]

**Review by ziyiz-amzn — APPROVED** [review]

**Pull request merged**